### PR TITLE
Fix area for non-polygons

### LINKED ISSUE #2305: Area gives wrong result for non-polygons
`area(::LazySet)` should convert to `VPolygon` to ensure that the vertices are sorted.

```julia
julia>  area(BallInf(zeros(2), 1.0))

0.0

julia> volume(BallInf(zeros(2), 1.0))
4.0
```

### PR DESCRIPTION
**Pull request by mforets**
Closes #2305.

### CODE REVIEW

File: src/convert.jl [hunk 1]
```diff
@@ -130,6 +130,32 @@ function convert(::Type{VPolygon}, P::AbstractPolytope)
...
+### Algorithm
+
+First we concretize `X` obtaining a (possibly) new representation `Xc`; then we
+wrap the result into a polygon in vertex representation by computing the list of
+vertices of `Xc`. The vertices of the resulting polygon in vertex representation
+are guaranteed to be sorted in counter-clockwise fashion.
+"""
+function convert(::Type{VPolygon}, X::LazySet)
+    @assert dim(X) == 2 "set must be two-dimensional for conversion"
+    return VPolygon(vertices_list(concretize(X)))
```
Comment (mforets) [inline src/convert.jl:156]: @schillic what do you think about using `concretize` here?

Comment (mforets) [reply src/convert.jl:156]: i would prefer to not use it, but note that, for example,

```julia
julia> vertices_list(rand(2, 10) * rand(Zonotope, dim=10))
ERROR: AssertionError: package 'Polyhedra' not loaded (it is required for executing `remove_redundant_vertices`)
Stacktrace:
 [1] macro expansion at /home/mforets/.julia/dev/LazySets/src/Assertions/Assertions.jl:23 [inlined]
....

julia> vertices_list(concretize(rand(2, 10) * rand(Zonotope, dim=10)))
36-element Array{Array{Float64,1},1}:
 [-31.25354193606883, -22.148084939907815]
 [-30.664022201688585, -23.27061463165326]
...
```

Comment (schillic) [reply src/convert.jl:156]: Agreed, I would delegate this decision to `vertices_list` instead. We could let `vertices_list` use `concretize`, possibly only if `Polyhedra` is not loaded.

Comment (mforets) [inline src/convert.jl:156]:
```suggestion
    return VPolygon(vertices_list(X))
```

File: src/convert.jl [hunk 2]
```diff
@@ -130,6 +130,32 @@ function convert(::Type{VPolygon}, P::AbstractPolytope)
...
+### Output
+
+The 2D set represented as a polygon.
+
+### Algorithm
+
+First we concretize `X` obtaining a (possibly) new representation `Xc`; then we
+wrap the result into a polygon in vertex representation by computing the list of
+vertices of `Xc`. The vertices of the resulting polygon in vertex representation
+are guaranteed to be sorted in counter-clockwise fashion.
```
Comment (schillic) [inline src/convert.jl:152]: The docstring is outdated.
```suggestion
We compute the list of vertices of `X` and wrap the result in a polygon in
vertex representation, which guarantees that the vertices are sorted in
counter-clockwise fashion.
```

**Pull request merged**